core/src/main/java/net/sourceforge/jnlp/util/OptionsDefinitionsPrinter.java

```diff
@@ -0,0 +1,149 @@
+import java.util.Arrays;
+import java.util.List;
+
+import static net.adoptopenjdk.icedteaweb.option.OptionsDefinitions.OPTIONS;
+
+/**
+ * This class allows to print the various set of options.
+ */
+public class OptionsDefinitionsPrinter {
+
+    public static List<OPTIONS> getItwsettingsCommands() {
+        return Arrays.asList(new OPTIONS[]{
+            OPTIONS.HELP2,
+            OPTIONS.LIST,
+            OPTIONS.GET,
+            OPTIONS.INFO,
+            OPTIONS.SET,
+            OPTIONS.RESET,
+            OPTIONS.RESETALL,
+            OPTIONS.HEADLESS,
+            OPTIONS.CHECK,
+            OPTIONS.VERBOSE
+        });
+    }
+
+    public static List<OPTIONS> getPolicyEditorOptions() {
+        return Arrays.asList(new OPTIONS[]{
+            OPTIONS.HELP1,
+            OPTIONS.FILE,
+            OPTIONS.DEFAULTFILE,
+            OPTIONS.CODEBASE,
+            OPTIONS.SIGNEDBY,
+            OPTIONS.PRINCIPALS,
+            OPTIONS.VERBOSE
+            }
+        );
+    }
+
+    public static List<OPTIONS> getJavaWsControlOptions() {
+        return Arrays.asList(new OPTIONS[]{
+            OPTIONS.ABOUT,
+            OPTIONS.VIEWER,
+            OPTIONS.CLEARCACHE,
+            OPTIONS.LISTCACHEIDS,
+            OPTIONS.LICENSE,
+            OPTIONS.HELP1}
+        );
+    }
+
+    public static List<OPTIONS> getJavaWsRuntimeOptions() {
+        return Arrays.asList(new OPTIONS[]{
+            OPTIONS.VERSION,
+            OPTIONS.ARG,
+            OPTIONS.PARAM,
+            OPTIONS.PROPERTY,
+            OPTIONS.UPDATE,
+            OPTIONS.VERBOSE,
+            OPTIONS.NOSEC,
+            OPTIONS.NOUPDATE,
+            OPTIONS.HEADLESS,
+            OPTIONS.STRICT,
+            OPTIONS.XML,
+            OPTIONS.REDIRECT,
+            OPTIONS.NOFORK,
+            OPTIONS.NOHEADERS,
+            OPTIONS.OFFLINE,
+            OPTIONS.TRUSTNONE,
+            OPTIONS.JNLP,
+            OPTIONS.HTML,
+            OPTIONS.BROWSER
+        });
+    }
+
+    public static List<OPTIONS> getJavaWsOptions() {
+        final List<OPTIONS> l = new ArrayList<>();
+        l.addAll(getJavaWsRuntimeOptions());
+        l.addAll(getJavaWsControlOptions());
+        //trustall is not returned by getJavaWsRuntimeOptions
+        //or getJavaWsControlOptions, as it is not desired in documentation
+        l.add(OPTIONS.TRUSTALL);
+        return l;
+    }
+
+    public static void main(String[] args) throws IOException {
+        switch (args[0]) {
+            case TextsProvider.JAVAWS:
+                printOptions(getJavaWsOptions());
+                break;
+            case TextsProvider.ITWEB_SETTINGS:
+                printOptions(getItwsettingsCommands());
+                break;
+            case TextsProvider.POLICY_EDITOR:
+                printOptions(getPolicyEditorOptions());
+                break;
+            default:
+                break;
+        }
+    }
+
+    private static void printOptions(final List<OPTIONS> options) {
```

Comment:
null check for options

Reply:
now you changed the functionality of the method.

I would just add Objects.requireNotNull(options); as first line of the method. if options is null a NPE would be thrown in the old code somewhere in the logic. By adding this line NPEs are much easier to find.

Reply:
done